Controllers/Backend/Mailarchive.php

```diff
@@ -97,15 +101,62 @@ public function downloadAction()
 
         if ($emlData) {
             $response->setHeader('Content-Type', 'application/octet-stream');
-            $response->setHeader('Content-Disposition',
-                'attachment; filename="' . $eml->getCreated()->format('Y-m-d His') . ' ' . $eml->getSubject() . '.eml"');
+            $response->setHeader(
+                'Content-Disposition',
+                'attachment; filename="' . $eml->getCreated()->format('Y-m-d His') . ' ' . $eml->getSubject() . '.eml"'
+            );
 
             $this->response->setBody($emlData);
         } else {
             $this->response->setBody('no eml-data saved');
         }
     }
 
+    /**
+     * Resend email
+     */
+    public function resendAction()
+    {
+        $this->Front()->Plugins()->ViewRenderer()->setNoRender();
+
+        $mailId = $this->Request()->getParam('id');
+        $mail = $this->getModelManager()->find(Mails::class, $mailId);
+
+        $htmlData = $mail->getBodyHtml();
+        $textData = $mail->getBodyText();
+        $attachments = $mail->getAttachments();
+        $sender = $mail->getSenderAddress();
+        $recieverString = $mail->getReceiverAddress();
+        $recievers = explode(',', $recieverString);
+
+        if ($textData) {
+            $mailObj = clone $this->container->get('mail');
+            $mailObj->setFrom($sender);
+            foreach ($recievers as $reciever) {
+                $mailObj->addTo($reciever);
```

Comment:
Hey, Thank you! :-)

This could result in showing the customer a BCC-Address 🤔 Maybe we should just use the first address, or adding a prompt, the user who wants to resend specifying the email-address 🤔

Reply:
Oops, good catch, you are right, user may have emails with BCC configured and the Mailarchive catches all of them ... The first address only may not be enough, so I guess a prompt would be the good ... but what if the user wants to send it identical to the original mail, including BCC addresses? Then he would have to specify which addresses should be TO, CC and BCC somehow.
I guess the "cleanest" way to store TO, CC and BCC would be in different DB fields or with a "marker", e.g. "TO:foo@bar.com,TO:bar@foo.com,CC:tom@jerry.com,BCC:homer@simpsons.com" etc.? 
Already existing archive emails would be treated as if they all have only TO addresses, newly archived emails would be splitted in TO, CC and BCC. 
What do you think?

Reply:
I would be fine with it.